### PR TITLE
fix([sdk][typescript]): isValidPath is not conform to BIP44

### PR DESCRIPTION
**Pull request by zouguangxian**
### Description
isValidPath is not conform to BIP44.

path level should be 

`m / purpose' / coin_type' / account' / change / address_index`

please refer to : https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4818)

### CODE REVIEW

File: ecosystem/typescript/sdk/src/aptos_account.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export class AptosAccount {
    * Test derive path
    */
   static isValidPath = (path: string): boolean => {
-    if (!/^m\/44'\/637'\/[0-9]+'\/[0-9]+'\/[0-9]+'+$/.test(path)) {
+    if (!/^m\/44'\/637'\/[0-9]+'\/[0-9]+\/[0-9]+$/.test(path)) {
```
Comment (alinush) [inline ecosystem/typescript/sdk/src/aptos_account.ts:41]: @kent-white, is this in line with our 1:00am call from a month ago discussing these apostrophes? Trying to cache all of that back in...

Comment (zouguangxian) [reply ecosystem/typescript/sdk/src/aptos_account.ts:41]: Is there any progress?

Comment (jjleng) [reply ecosystem/typescript/sdk/src/aptos_account.ts:41]: This was intentional, Near and Solana have apostrophes for all components.

Comment (zouguangxian) [reply ecosystem/typescript/sdk/src/aptos_account.ts:41]: In the current implementation, with/without apostrophes will generate the same keys.

BIP44 is mentioned multiple times in the documents and codes.  I just figured out the format like `m/44'/637'/0'/0'/1'` is not conform BIP44, it should be `m/44'/637'/0'/0/1` if it does.

The format is defined in BIP44, is it not related to ED25519?

Comment (alinush) [reply ecosystem/typescript/sdk/src/aptos_account.ts:41]: Here's some context to understand the use of Ed25519-based HD wallets (via [BIP-32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki), [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki) and [SLIP-10](https://github.com/satoshilabs/slips/blob/master/slip-0010.md))

- See more details in [this medium post](https://medium.com/@robbiehanson15/the-math-behind-bip-32-child-key-derivation-7d85f61a6681)
- First time Ed25519 was asked to be used with BIP32: https://github.com/satoshilabs/slips/issues/39
    - BIP32 seems to assume there exists a linear homomorphism $\phi(sk)\rightarrow pk$ such that $\phi(sk_1)\cdot\phi(sk_2)=\phi(sk_1+sk_2)$ for any secret keys $(sk_1,sk_2)$
        - In more plain terms: $sk'=sk+\delta\Rightarrow pk'=pk\cdot g^\delta$
        - This does not work for Ed25519 where $pk = g^{H(sk)}$ for some hash function $H$
        - See this [StackExchange answer](https://crypto.stackexchange.com/q/31369/21296)
        - See [this paper](https://ieeexplore.ieee.org/document/7966967/?reload=true) for a potential fix

**Pull request closed without merging**